libs/src/services/discoveryProvider/index.js

```diff
@@ -722,11 +730,9 @@ class DiscoveryProvider {
           !indexedBlock ||
           (chainBlock - indexedBlock) > UNHEALTHY_BLOCK_DIFF
         ) {
-          // Clear any cached discprov
-          localStorage.removeItem(DISCOVERY_PROVIDER_TIMESTAMP)
           // Select a new one
           console.info(`${this.discoveryProviderEndpoint} is too far behind, reselecting discovery provider`)
-          const endpoint = await this.autoSelectEndpoint()
+          const endpoint = await this.autoSelectEndpoint(AUTOSELECT_DISCOVERY_PROVIDER_RETRY_COUNT, true)
```

Comment:
@raymondjacobson @vicky-g, just two things I want to make sure of here:
- do we ever eventually fail the request? or if we hit 0 retries, do we round-robin forever? (obviously don't want to do that)
- It looks like we're potentially reselecting an endpoint twice in this function - once if it fails, and then we throw, retry, and potentially select again. Does this have any weird effects (e.g. skipping a DP), and would it make sense to consolidate the autoselect logic in a single point in this method?

Reply:
1-- we should probably add a failover case if we've tried everything, but also that means all of audius discovery providers is down, so it's just a really unfortunate situation. maybe we dn't have to do this now? but wouldn't hurt.

2-- subtle behavior here. the first time we select it's due to the cached version OR it's because we picked using just the /version endpoint. Autoselect does health check on each of the endpoints, so we shouldn't pick a bad one, but in the case that we do, we'd just try again. so long as /health_check is coming back OK though, i think it's reasonable that we select it, we just have to be better about making /health_check good

@vicky-g thoughts?

Reply:
conversation that occurred offline:
1. will add a global max retry count
2. surface level there doesn't seem to be any weird side effects. did reset retry count to ensure that the recursive call will occur